src/smf/packet-smf-compress.c

```diff
@@ -136,15 +153,21 @@ DIAG_ON(cast-qual)
             outStr = "Unknown";
             break;
         }
-        g_print("ssl_decompress_record: Frame %d inflate() failed (%d): %s: %s\n", pinfo->fd->num, err, outStr, stream_p->msg);
+        
+        snprintf(errorMsg_p, ERROR_MSG_SIZE, "%s(%d): %s", outStr, err, stream_p->msg);
+        *outl = *outl - stream_p->avail_out;
+        
+        // Cleanup the decompressor and re-initialize
+        inflateEnd(&compressed_stream_p->stream);
+        init_stream(compressed_stream_p);
```

Comment:
This looks strange, it seems like we always have a stream initialized

Reply:
When processing a compressed pcap that is also snaplen (the amount of data for each frame is reduced during capture). The decompression engine will fail. If I do not reset, the rest of the file cannot be decompressed. The only way around it is to start over again.
Doing the above reset allows us to continue to see some data (especially if you are interested in the topic of the smf message).
I do not know of a way to tell the decompressor that certain bytes are missing in between frames, so this is my only option.